src/main/webapp/app/shared/util/core-drive-annotation-submission/core-drive-annotation-submission.ts

```diff
@@ -0,0 +1,136 @@
+    const tempTumors: Tumor[] = [];
+    if (shouldExclude(onlyReviewedContent, mutation.name_review)) {
+      tempMutations.push(mutation);
+      continue;
+    }
+    processData(mutation, ['name'], excludeComments, onlyReviewedContent);
+    processData(mutation.mutation_effect, ['oncogenic', 'effect', 'description'], excludeComments, onlyReviewedContent);
+    for (const tumor of mutation.tumors ?? []) {
+      if (shouldExclude(onlyReviewedContent, tumor.cancerTypes_review)) {
+        tempTumors.push(tumor);
+        continue;
+      }
+      // process tumor cancerTypes
+      processData(tumor, ['summary', 'diagnosticSummary', 'prognosticSummary'], excludeComments, onlyReviewedContent);
+      processData(tumor.diagnostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      processData(tumor.prognostic, ['level', 'description', 'relevantCancerTypes'], excludeComments, onlyReviewedContent);
+      for (const ti of tumor.TIs) {
+        processData(ti, ['description'], excludeComments, onlyReviewedContent);
+        type TempTreatment = Omit<Treatment, 'name'> & { name: ReturnType<typeof drugUuidToDrug> | string };
+        const tempTreatments: TempTreatment[] = [];
+        for (const treatment of ti.treatments ?? []) {
+          if (shouldExclude(onlyReviewedContent, treatment.name_review)) {
+            tempTreatments.push(treatment);
+            return true;
+          }
+          (treatment as TempTreatment).name = drugUuidToDrug(treatment.name, drugList);
+          processData(
+            treatment,
+            ['level', 'propagation', 'propagationLiquid', 'indication', 'description'],
+            excludeComments,
+            onlyReviewedContent,
+          );
+        }
+        for (const item of tempTreatments) {
+          const index = ti.treatments.indexOf(item as Treatment);
+          if (index !== -1) {
+            ti.treatments.splice(index, 1);
+          }
+        }
+      }
+    }
+    for (const tumor of tempTumors) {
+      const index = mutation.tumors.indexOf(tumor);
+      if (index !== -1) {
+        mutation.tumors.splice(index, 1);
+      }
+    }
+  }
+  for (const mutation of tempMutations) {
+    const index = gene.mutations.indexOf(mutation);
+    if (index !== -1) {
+      gene.mutations.splice(index, 1);
+    }
+  }
+  return gene;
+}
+
+function processData(data: object | undefined, keys: string[], excludeComments: boolean, onlyReviewedContent: boolean) {
+  if (data !== undefined) {
+    for (const key of keys) {
+      if (excludeComments) {
+        delete data[key + '_comments'];
+      }
+      const reviewKey = key + '_review';
+      const maybeReview = data[reviewKey];
+      if (onlyReviewedContent && typeof maybeReview === 'object' && maybeReview !== null && 'lastReviewed' in maybeReview) {
+        data[key] = maybeReview.lastReviewed;
+      }
+    }
+  }
+}
+
+export function getVUSData(vus: Vus[], excludeComments: boolean) {
```

Comment:
We haven't needed to include comments for parsing in server side for a long time. Let's remove.

Reply:
My apology, could've saved you some time for writing these tests.

Reply:
no worries